<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Lei Li leil@uchicago.edu" />

<meta name="date" content="2019-07-02" />

<title>Librator: effcient and effective protein expression by using Gibson Cloning</title>
<link rel="icon" type="image/x-icon" href="logo.ico"/>

<script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.8.0.js"></script>

<link rel="stylesheet" type="text/css" href="css/style.css">


<script type="text/javascript"> 
function showorhide()  
{  
  var mychar = document.getElementById("con").style.display;
  if (mychar == "none") 
  {
    document.getElementById("con").style.display ="block";  
  }
  else
  {
    document.getElementById("con").style.display ="none";
  }
}
</script> 

<style>
div.navbox {
    width: 20%;
}

.menu {
  position: fixed;
  right: 0px;
  width: 10%;
  background-color: silver;
  height: 100%;
  top: 56px;
}

.info {
  position: absolute;
  background-color: white;
  left: 20%;
  width: 75%;
  height: 100%;
  z-index: 999;
  top: 56px;
  padding-left: 30px;
}

.img {
  box-shadow: 10px 10px 5px #888888;
}

.left{
  position: fixed;
  left: 0px;
  width: 20%;
  background-color: #f5f5f5;
  height: 100%;
  top: 56px;
}

.author_name{display:block;color:#F36170;font-size:2em}
.author_bio{font-size:1.2em;display:block}
.author_job{display:block;color:#8d7edc;font-size:1.35em;font-weight:700;text-transform:uppercase;margin:5px auto 10px}

.title {
  background: #f0fbeb;
  border: 1px solid #9bdf70;
  border-radius: 5px;
  display:block;
  width: 90%;
}

a{
  color: black;
}

li{
  margin-top: 4px;
}

.text {
  display: none;
  width: 90%;
}
.img{
  display: none;
}

.center {
  vertical-align: middle;
  text-align: center;
  display: inline-block;
  margin-top: 50%
}

.center span{
  text-align: right;
}

</style>

<script type="text/javascript">
$(document).ready(function(){
  $('.title').click(function(){
    var father = $(this).parent('li');
    var display = father.children(".text").css('display');
    if(display == 'none'){
      father.children(".text").css('display','block');
      father.children(".img").css('display','block');
    } else {
      father.children(".text").css('display','none');
      father.children(".img").css('display','none');
    }
  });
});
</script>

</head>

<body>


<div class="navbox">
<ul class="nav">
  <li><a href="index.html">About</a></li>
  <li><a href="install.html">Install</a></li>
  <li><a class="active" href="tutorial.html">Tutorial</a></li>
  <li><a href="contact.html">Contact</a></li>
</ul>
</div>

<div class="row">

  <div class="left">
    <div class="center">
      <img class="logo" src="img/logo.png" style="margin:auto;">
      <span class="author_name">Librator: effcient and effective protein expression by using Gibson Cloning</span>
      <span class="author_bio">Wilson Lab</span>
      <span class="author_job">The University of Chicago </span>
    </div>
  </div>
  <div class="info">
    <h1>User Interface</h1>
    <ul>
      <li>
        <a class="title">Main Tab</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/MainTab.png" style="width: 50%">
      </li>
      <li>
        <a class="title">Sequence Tab</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/SequenceTab.png" style="width: 50%">
      </li>
      <li>
        <a class="title">Alignment RTF Tab</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/AlignmentRTFTab.png" style="width: 50%">
      </li>
      <li>
        <a class="title">Alignment Tab (HTML)</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/AlignmentHTML.png" style="width: 90%">
      </li>
      <li>
        <a class="title">Sequence DB Tab</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/SequenceDBTab.png" style="width: 90%">
      </li>
      <li>
        <a class="title">Summary Tab</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/SummaryTab.png" style="width: 90%">
      </li>
      <li>
        <a class="title">Summary Tab (HTML)</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/SummaryHTMLTab.png" style="width: 90%">
      </li>
      <li>
        <a class="title">Fragment DB Tab</a>
        <p class="text">Users can set paths for all required tools and databases.</p>
        <img class="img" src="img/FragmentDBTab.png" style="width: 90%">
      </li>
    </ul>
    <h1>Functions</h1>
      <h2>Basic functions</h2>
      <ul>
        <li>
          <a class="title">Path setting</a>
          <p class="text">Users can set paths for all required tools and databases. Click Setting-> Preferences in menu</p>
          <img class="img" src="img/PathSet.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Create new sequence database</a>
          <p class="text">Users can create new sequence database. Click File-> New in menu </p>
          <img class="img" src="img/newDB.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Open existing sequence database</a>
          <p class="text">Users can create new sequence database. Click File-> Open in menu </p>
          <img class="img" src="img/openDB.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Import sequences</a>
          <p class="text">After load an existing sequence database, users can import sequence into current database. The input files should be in FASTA format. Click Sequences-> Import in menu</p>
          <img class="img" src="img/importSeq.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Export sequences</a>
          <p class="text">After load an existing sequence database, users can import sequence into current database. Selected sequences will be exported to a FASTA file. Click Sequences-> Export in menu </p>
          <img class="img" src="img/exportSeq.png" style="width: 50%">
        </li>
      </ul>

      <h2>Advanced functions</h2>
      <ul>
        <li>
          <a class="title">Edit information of sequences (on Main tab)</a>
          <p class="text">Users can edit information (sequence name, Role, Form, Subtype, NT sequence, reading frame, donor regions) of sequences on main tab.</p>
          <p class="text">For Sequence name, users can click sequence name text input to edit it.</p>
          <img class="img" src="img/editSeqName.png" style="width: 50%">
          <p class="text">For Role, Form, and Subtype, users can edit by the left panel. Users can select multiple sequences in active sequence panel to do bulk update.</p>
          <img class="img" src="img/changeRole.png" style="width: 50%">
          <p class="text">For reading frame start and end, users can use the “Coding region” inputs. For NT sequences, users can click “edit sequence” button to edit sequences.</p>
          <img class="img" src="img/updateSeq.png" style="width: 50%">
          <p class="text">Note: coding region and NT sequence editing is disabled for sequences with mutation information. For donor regions information, users can click “Donor region” input to update information.</p>
        </li>
        <li>
          <a class="title">HA numbering</a>
          <p class="text">Users can access H1/H3 numbering of selected sequence. Click Tools-> HA Numbering in menu or just click “Sequence” tab</p>
          <img class="img" src="img/SequenceTab.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Multiple sequence alignment</a>
          <p class="text">Users can align multiple sequences together and check the results in a graphical viewer. Click Tools-> Multiple Alignment(HTML) in menu. There are two display modes: original sequence mode and template mode. In template mode, users can choose any sequence (including consensus sequence) as template to only highlight sequence difference.</p>
          <img class="img" src="img/MSA.png" style="width: 90%">
        </li>
        <li>
          <a class="title">Generate phylogenetic tree (ML tree)</a>
          <p class="text">Users can generate phylogenetic tree (Maximum Likelihood tree) for selected sequences. Nucleotide and Amino Acid trees are available. For Nucleotide tree: Click Tools-> Generate Maximum Likelihood Tree (nucleotide) in menu; For Amino acid tree: Click Tools-> Generate Maximum Likelihood Tree (Amino Acid) in menu. </p>
          <img class="img" src="img/tree.png" style="width: 90%">
        </li>
        <li>
          <a class="title">Mutate sequence</a>
          <p class="text">Users can generated new sequences with mutations based on any existing sequence. Click Edit Sequence-> Mutation in menu. As shown below, users can generate mutations on a template sequence using either original position (count from first amino acid, panel A) or H1/H3 numbering (H1 numbering for group1 virus and H3 numbering for group2 virus, panel B). H1/H3 numbering function is only enabled for HA sequences of FLU A. Users are allowed to input multiple mutations at the same time, and are allowed to decide from two options: 1) generate one sequence with all mutations; 2) generate multiple sequences, each of them with a single mutation. After the mutated sequence was generated, users can visualize the 3D structure (only enabled for HA now) of generated HA protein (panel C).</p>
          <img class="img" src="img/MUTATION.png" style="width: 90%">
        </li>
        <li>
          <a class="title">Compare sequences and generate screening mutations</a>
          <p class="text">Users can compare sequences and generate consensus sequences (Base biased) or new sequences with screening mutations (Cocktail). Click Edit Sequence-> Editing in menu. Cocktail mode (A): Compare base sequence to another sequence, identify all mutations between two sequences on donor region of donor sequence (user can setup donor region on this interface), then generate new sequences with those mutations using base sequence as template. Users can choose to only generate sequences with single mutations or all combinations of mutations. Base biased mode (B): Generate consensus sequences of base sequence and selected sequences. For each donor sequence, only their donor region will be considered (full sequence will be considered when donor region is none).</p>
          <img class="img" src="img/edit.png" style="width: 90%">
        </li>
        <li>
          <a class="title">Epitope transplant across different subtypes</a>
          <p class="text">Users can transplant epitopes/regions from sequences of different subtypes. Regions being removed from base sequence and that being inserted from donor sequence are not necessary to have same length. Users can add multiple replacements on one sequence (click ‘Add’). Those replacements could be from different sequences. Users can review the replacement design in the bottom window. Users can double click replacement to delete it. Users also can click ‘Clear’ button to rest current design. Click Edit Sequence-> Fusion in menu. This window has two layouts: high resolution layout and low resolution layout. Liberator will automatically choose the best fit based on current display resolution. Users also can click Edit Sequence-> Fusion (high resolution) or Edit Sequence-> Fusion (low resolution) to determine the layouts they prefer (As shown below).</p>
          <img class="img" src="img/fusionH.png" style="width: 90%">
          <img class="img" src="img/fusionL.png" style="width: 50%">
        </li>
        <li>
          <a class="title">3D visualization via PyMOL</a>
          <p class="text">Users can see 3D structure of selected sequence via PyMOL (only for HA). Click Visualization-> PyMOL in menu. </p>
          <img class="img" src="img/PyMOL.png" style="width: 90%">
        </li>
        <li>
          <a class="title">Generate Gibson Clone fragments</a>
          <p class="text">Users can generate Gibson Clone Fragments. Users can select sequences, determine fragment DB, output path, and joint region plan (panel A), after click “Generate Fragments” button, a dialog will pop up for users to review and confirm (panel B). On panel B, users can review the generated Fragments (both amino acid sequences and nucleotide sequences), joint regions that shared by neighbor fragments are highlighted. Joint region that connect fragment 1 and 4 to the vector not displayed for AA sequences but can be found in NT sequences. ‘-’ in AA alignments indicate incomplete sequences or deletions and will be deleted in NT fragments. Users can click “confirm” button to generate fragments after they confirmed current fragments. After a few second, a notice will be popped up with summarized results (panel C).</p>
          <img class="img" src="img/Gibson.png" style="width: 50%">
        </li>

      </ul>
  </div>



</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
